gsw_check_functions.c

```diff
@@ -97,12 +97,12 @@ double          tf_poly[cast_m*cast_n];
 double          h[cast_m*cast_n];
 double          z[cast_m*cast_n];
 
-int assert_equal(int actual, int expected, const char *test_name) {
+void assert_equal(int actual, int expected, const char *test_name) {
```

Comment:
If you want we can merge a PR with this fix and work on the makefile in anothter PR. Also, maybe we don't need a full test matrix with all OSes. I guess using the C++ approach is important for Windows only and we can test if there.